internal/flushcommon/pipeline/data_sync_service.go

```diff
@@ -362,19 +325,39 @@ func getServiceWithChannel(initCtx context.Context, params *util.PipelineParams,
 // NewDataSyncService stops and returns the initCtx.Err()
 func NewDataSyncService(initCtx context.Context, pipelineParams *util.PipelineParams, info *datapb.ChannelWatchInfo, tickler *util.Tickler) (*DataSyncService, error) {
 	// recover segment checkpoints
-	unflushedSegmentInfos, err := pipelineParams.Broker.GetSegmentInfo(initCtx, info.GetVchan().GetUnflushedSegmentIds())
-	if err != nil {
-		return nil, err
+	var (
+		err                   error
+		metaCache             metacache.MetaCache
+		unflushedSegmentInfos []*datapb.SegmentInfo
```

Comment:
In previous implement, if the len(unflushedSegmentIds) == 0. unflushedSegmentInfos will be a empty slice but not a nil.
Please verify if a nil slice here is safe.